push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/AdmPushNotificationSender.java

```diff
@@ -0,0 +1,63 @@
+package org.jboss.aerogear.unifiedpush.message.sender;
+
+
+import org.jboss.aerogear.unifiedpush.api.AdmVariant;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+
+import org.jboss.aerogear.unifiedpush.message.UnifiedPushMessage;
+import org.jboss.aerogear.unifiedpush.utils.AeroGearLogger;
+import org.jboss.aerogear.adm.ADM;
+import org.jboss.aerogear.adm.AdmService;
+import org.jboss.aerogear.adm.PayloadBuilder;
+
+import java.util.Collection;
+import java.util.Set;
+
+@SenderType(AdmVariant.class)
+public class AdmPushNotificationSender implements PushNotificationSender {
+    private final AeroGearLogger logger = AeroGearLogger.getInstance(AdmPushNotificationSender.class);
+
+    @Override
+    public void sendPushMessage(Variant variant, Collection<String> clientIdentifiers, UnifiedPushMessage pushMessage, NotificationSenderCallback senderCallback) {
+        AdmService admService = ADM.newService();
+
+        PayloadBuilder builder = ADM.newPayload();
+
+        //flatten the "special keys"
+        builder.dataField("alert", pushMessage.getMessage().getAlert());
+        builder.dataField("sound", pushMessage.getMessage().getSound());
+        builder.dataField("badge", "" + pushMessage.getMessage().getBadge());
```

Comment:
@sebastienblanc Can you create a JIRA for the 'push-message-format' update? I think the new payload from @edewit for Windows was also never added to that 'spec' doc